### ISSUE TITLE
[Documentation Issue] Where did crypto installation instructions go?

### ISSUE THREAD

**Issue by jgeewax** [GoogleCloudPlatform/gcloud-python#1790]
Page Name: index
Release: 0.12.0

The landing page lost some of the install troubleshooting instructions. Any idea where that went?

> If you have trouble installing pycrypto or pyopenssl (and you’re on Ubuntu), you can try install the precompiled packages:
> $ sudo apt-get install python-crypto python-openssl
> If you want to install everything with pip, try installing the dev packages beforehand:
> $ sudo apt-get install python-dev libssl-dev
> If you want to install gcloud-python from source, you can clone the repository from GitHub:
> $ git clone git://github.com/GoogleCloudPlatform/gcloud-python.git
> $ cd gcloud-python
> $ python setup.py install

/cc @rok987

**Comment by dhermes** [comment]
We don't have any crypto dependencies and now `oauth2client` [works just fine][1] with pure Python `rsa`, `pyasn1` and `pyasn1_modules`. ([ref][2])

[1]: https://github.com/google/oauth2client/blob/e99a4ac3c5d3c2944680ea180b71a0d144d6128b/oauth2client/crypt.py#L70-L71
[2]: https://github.com/google/oauth2client/blob/e99a4ac3c5d3c2944680ea180b71a0d144d6128b/oauth2client/_pure_python_crypt.py

**Comment by jgeewax** [comment]
OK - so those dependencies are 100% not needed anymore, yes?

**Comment by dhermes** [comment]
The only users that would ever need it are users with P12 key files for service accounts. If we come across such a user we will instruct them that JSON key files are the preferred, strongly encourage that they upgrade and then go about our day. Close this out if you don't have any issues?

**Comment by rok987** [comment]
I don't know if this is related or not (we can create a new issue if need be), but someone tried to use gcloud python on a CentOS 7 image (VM on GCE from my understanding) and in order to successfully use BigQuery/Datastore they had to do these instructions (copy-pasting what they sent):

```
sudo su -
rpm -iUvh http://dl.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-5.noarch.rpm
yum install python-pip

curl https://sdk.cloud.google.com | bash
gcloud init

yum install python-devel
yum install libffi-devel
yum install openssl-devel

yum group install "Development Tools"

pip install --upgrade gcloud
```

Is that something you are aware of? There might be better ways, but it's what worked for them. We should probably document that somewhere. Also, instructions might be different on other images.

**Comment by dhermes** [comment]
Most of those are outside of our scope.

-   Adding the extra packages repo (EPEL) is [required][1] for `pip`, however having a valid Python install isn't something a project would document, that would be something for python.org

    ```
    rpm -iUvh http://dl.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-5.noarch.rpm
    yum install python-pip
    ```

-   Installing the [`gcloud` CLI][2] is not necessary to use our library, **especially** on GCE. However we do provide links to it in our docs and any install instructions for that should lie within the `gcloud` CLI docs

    ```
    curl https://sdk.cloud.google.com | bash
    gcloud init
    ```

-    I'm not sure if we need Python headers for native extensions, so `yum install python-devel` may be necessary? But this is somewhat a "have a valid Python install" requirement

-    We no longer require OpenSSL, hence `libffi-devel` (which is required by `openssl-devel`) is not needed and it's possible they added `python-devel` for these reasons

    ```
    yum install libffi-devel
    yum install openssl-devel
    ```

-   I'm not sure what `yum group install "Development Tools"` does

[1]: http://www.liquidweb.com/kb/how-to-install-pip-on-centos-7/
[2]: https://cloud.google.com/sdk/gcloud/

**Comment by jgeewax** [comment]
CentOS 7 should be pretty straight forward, and users of CentOS / RHEL should know how to get PIP installed. I just tried this on CentOS7, and it went like this:

```sh
$ sudo yum install epel-release  # This is Extended Packages for Enterprise Linux -- very standard.
$ sudo yum install python-pip  # Worked just fine
$ pip install gcloud  # Worked just fine
```

I then tested that I could at least auth and make requests to our Cloud APIs, which seemed to work just fine...

We have to have *some* level of "minimum knowledge" here -- otherwise we're going to start documenting how to install Python... which is... way out of scope.

I was in agreement that if one of our dependencies requires something strange (as we had with the crypto packages) then we should have it in the FAQ to explain what's up -- but documenting how to install Python and PIP on every OS seems too far off the mark.
